util/plot_utils.py

```diff
@@ -4,20 +4,52 @@
 """
 import torch
 import pandas as pd
-from pathlib import Path
 import seaborn as sns
 import matplotlib.pyplot as plt
 
+from pathlib import Path, PurePath
+from typing import Iterable
+import copy
 
-def plot_logs(logs, fields=('class_error', 'loss_bbox_unscaled', 'mAP'), ewm_col=0):
-    dfs = [pd.read_json(Path(p) / 'log.txt', lines=True) for p in logs]
+
+def plot_logs(logs, fields=('class_error', 'loss_bbox_unscaled', 'mAP'), ewm_col=0, log_name='log.txt'):
+    '''
+    Function to plot specific fields from training logs
+    :: Inputs - logs = list containing one or more directories, each w/ single log file (convert to list if needed)
+              - fields = which training results to plot from log file
+              - ewm_col = ??
+              - log_name = optional, name for log file(s) if different than default 'log.txt'
+    :: Outputs - matplotlib plots of items in fields per each log file.
+               - solid lines are training results, dashed lines are test results
+    '''
+
+    # verify logs is list, convert if not. handle single dir, etc. *Should not have side-effects, makes deep new logs
+    if not isinstance(logs, list):
+        if not isinstance(logs, Iterable):
+            logs = [logs]
```

Comment:
I think this could lead to subtle bugs.

If the user passes a single string (which is an Iterable), it will go in the next case, so converting it into a list of single-character strings.